compiler/src/dmd/tokens.d

```diff
@@ -946,16 +946,17 @@ nothrow:
 
     extern (C++) const(char)* toChars() const
     {
-        __gshared char[3 + 3 * floatvalue.sizeof + 1] buffer;
+        const bufflen = 3 + 3 * floatvalue.sizeof + 1;
+        __gshared char[bufflen] buffer;
         const(char)* p = &buffer[0];
         switch (value)
         {
         case TOK.int32Literal:
-            sprintf(&buffer[0], "%d", cast(int)intvalue);
+            snprintf(&buffer[0], bufflen, "%d", cast(int)intvalue);
```

Comment:
Similarly
```suggestion
            snprintf(&buffer[0], buffer.length, "%d", cast(int)intvalue);
```

Reply:
yes, but here buffer.length is used more often so I wanted to avoid the indirection.

Reply:
What indirection? It's a compile time constant.

Reply:
Oh, yeah makes sense, I did not know that.